internal/metrics/metrics.go

```diff
@@ -0,0 +1,209 @@
+package metrics
+
+import (
+	"context"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"go.opencensus.io/stats"
```

Comment:
when is OTEL Collector planning to switch to OTEL-Go SDK? I am not a fan of depending on OpenCensus. Also, if you're implementing Jaeger's metrics.Factory anyway, could't we just use the normal one that we create? Or the issue is that the OTEL config does not allow to propagate that custom object through configs?

Reply:
Still blocked by the sdk https://github.com/open-telemetry/opentelemetry-collector/issues/4198

My intention was to have it consistent with the collector experience I don't believe we can anything to avoid this without opening a second port to scrape. There's no way for me to tell the collector, 'no use this instead'. It's still opencensus everywhere.

At least the dependency is only over here not in the main source.